lib/portlayer/exec/container.go

```diff
@@ -157,23 +159,15 @@ func (s State) String() string {
 
 // CurrentState returns current state.
 func (c *Container) CurrentState() State {
-	c.m.Lock()
-	defer c.m.Unlock()
-	return c.state
-}
-
-// SetState changes container state.
-func (c *Container) SetState(s State) State {
-	c.m.Lock()
-	defer c.m.Unlock()
-	return c.updateState(s)
+	v := atomic.LoadInt32((*int32)(&c.state))
+	return State(v)
 }
 
-func (c *Container) updateState(s State) State {
+func (c *Container) swapState(s State) State {
 	log.Debugf("Setting container %s state: %s", c.ExecConfig.ID, s)
 	prevState := c.state
 	if s != c.state {
-		c.state = s
+		atomic.StoreInt32((*int32)(&c.state), int32(s))
```

Comment:
This seems out of place here. The other elements of Container are not protected at all, so one would assume there are no locks acquired when `swapState` is called. That may well be the case, and if it is, please see if you can just replace this with assignment.